traffic_ops/traffic_ops_golang/servers.go

```diff
@@ -163,3 +167,266 @@ JOIN status st ON s.status = st.id
+			log.Error.Printf("could not scan ds fields row: %s\n",err)
+			return nil, DBError
+		}
+		if XmlId.Valid && len(XmlId.String)>0 {
+			param := "hdr_rw_" + XmlId.String + ".config"
+			if EdgeHeaderRewrite.Valid && len(EdgeHeaderRewrite.String)>0 {
+				insert = append(insert,param)
+			} else {
+				delete = append(delete,param)
+			}
+			param = "regex_remap_" + XmlId.String + ".config"
+			if RegexRemap.Valid && len(RegexRemap.String)>0 {
+				insert = append(insert,param)
+			} else {
+				delete = append(delete,param)
+			}
+			param = "cacheurl_" + XmlId.String + ".config"
+			if CacheUrl.Valid && len(CacheUrl.String)>0 {
+				insert = append(insert,param)
+			} else {
+				delete = append(delete,param)
+			}
+		}
+
+	}
+
+	//insert the parameters we selected above:
+	insertParams, err := tx.Prepare(`INSERT INTO parameter (config_file, name, value)
+	WITH
+	q1 AS (SELECT UNNEST($1::text[]) AS config_file),
+	q2 AS (SELECT * FROM (VALUES ($2) ) AS name),
+	q3 AS (SELECT * FROM (VALUES ($3) ) AS value)
+	 SELECT * FROM q1,q2,q3 ON CONFLICT DO NOTHING`)
+	if err != nil {
+		log.Error.Printf("could not prepare parameter bulk insert query: %s\n",err)
+		return nil, DBError
+	}
+	fileNamePqArray := pq.Array(insert)
+	_, err = insertParams.Exec(fileNamePqArray,"location",atsConfigLocation)
+	if err != nil {
+		log.Error.Printf("could not execute parameter bulk insert: %s\n",err)
+		return nil, DBError
+	}
+
+	//select the ids associated with the parameters we created above (may be able to get them from insert above to optimize)
+	selectParameterIds, err := tx.Prepare("SELECT id FROM parameter WHERE name = 'location' AND config_file IN ($1)")
+	if err != nil {
+		log.Error.Printf("could not prepare parameter id select query: %s\n",err)
+		return nil, DBError
+	}
+	rows, err = selectParameterIds.Query(fileNamePqArray)
+	if err != nil {
+		log.Error.Printf("could not execute parameter id select query: %s\n",err)
+		return nil, DBError
+	}
+	parameterIds := []int64{}
+	for rows.Next() {
+		var Id int64
+		if err := rows.Scan(&Id); err != nil {
+			log.Error.Printf("could not scan parameter id: %s\n",err)
+			return nil, DBError
+		}
+		parameterIds = append(parameterIds,Id)
+	}
+
+	//associate all parameter ids with the profiles associated with all servers associated with assigned dses.
+	insertProfileParams, err := tx.Prepare(`INSERT INTO profile_parameter (profile, parameter)
+	WITH
+	q1 AS ( SELECT DISTINCT profile FROM server LEFT JOIN deliveryservice_server ON server.id = deliveryservice_server.server WHERE deliveryservice_server.deliveryservice = ANY($1::bigint[]) ),
+	q2 AS (SELECT UNNEST($2::bigint[]) AS parameter)
+	SELECT * FROM q1,q2
+	ON CONFLICT DO NOTHING`)
+	if err != nil {
+		log.Error.Printf("could not prepare profile_parameter bulk insert: %s\n",err)
+		return nil, DBError
+	}
+	_, err = insertProfileParams.Exec(dsPqArray,pq.Array(parameterIds))
+	if err != nil {
```

Comment:
This can be collapsed, `if _, err = insertProfileParams.Exec(dsPqArray,pq.Array(parameterIds)); err != nil {`